cmd/gen-corim.go

```diff
@@ -141,39 +141,40 @@ func generate(attestation_scheme *string, evidence_file *string, key_file *strin
 		return err
 	}
 
-	measurements := GetMeasurementsFromComponents(schemeClaims.swComponents, schemeClaims.config, *attestation_scheme == "cca")
-
-	//creating a new reference value containing the measurements and the implementation ID from the evidence token
+	//creating new reference values containing the measurements and the implementation ID from the evidence token
 	class := comid.NewClassImplID(schemeClaims.implID)
 
-	refVal := comid.ReferenceValue{
-		Environment:  comid.Environment{Class: class},
-		Measurements: measurements,
+	refVals, err := GetRefValsFromComponents(schemeClaims, class, *attestation_scheme == "cca")
+	if err != nil {
+		_ = os.RemoveAll(dir)
```

Comment:
Just a note: this (and all the similar instances above and below) could be handled with a single [`defer`](https://go.dev/tour/flowcontrol/12) statement.